src/controller/OperationalCredentialsDelegate.h

```diff
@@ -67,6 +69,12 @@ class DLL_EXPORT OperationalCredentialsDelegate
                                         const ByteSpan & DAC, const ByteSpan & PAI,
                                         Callback::Callback<OnNOCChainGeneration> * onCompletion) = 0;
 
+    virtual CHIP_ERROR SignICAC(const ByteSpan & icaCsr, Callback::Callback<OnICACSigned> * onCompletion) = 0;
+
+    virtual CHIP_ERROR SignNOC(const ByteSpan & icac, const ByteSpan & nocCsr, MutableByteSpan & noc) = 0;
+
+    virtual CHIP_ERROR ObtainIcaCsr(MutableByteSpan & icaCsr) = 0;
```

Comment:
1. Why the differences in casing between NOC/ICAC and ICA and CSR? 
2. This seems to envision a setup where we sign an ICAC for some other entity in SignICAC, right?  Then how is SignNOC supposed to work?  I can't tell whether I am missing something or whether this API is not actually implementable, not least because there is no documentation for what any of these things are supposed to do.  So let's start by adding said documentation.

Reply:
All the changes in src/controller have been reverted.